eth/protocols/eth/handler.go

```diff
@@ -106,7 +107,9 @@ func MakeProtocols(backend Backend, network uint64, disc enode.Iterator) []p2p.P
 			Version: version,
 			Length:  protocolLengths[version],
 			Run: func(p *p2p.Peer, rw p2p.MsgReadWriter) error {
-				peer := NewPeer(version, p, rw, backend.TxPool())
+				txpool, allowed := backend.TxPool(p)
+				peer := NewPeer(version, p, rw, txpool)
+				peer.SetAllowedForTxGossip(allowed)
```

Comment:
with my suggestion, this can become
```suggestion
				peer := NewPeer(version, p, rw, backend.TxPool(p))
```
again